plugins/catalog-react/api-report.md

```diff
@@ -460,6 +460,7 @@ export type EntityRefLinksProps<
 
 // @public
 export interface EntityRefPresentation {
+  promise: Promise<EntityRefPresentationSnapshot>;
```

Comment:
maybe?

```suggestion
  promise(): Promise<EntityRefPresentationSnapshot>;
```

Reply:
The other ones weren't methods so I didn't make one here either for consistency. I considered maybe making it a getter anyway though behind the scenes for lazy triggering the load+render. May still do that at some point. But people can still consume it then as if it's a regular property.

Reply:
but why do we need to create promises (or observables or others things) if they are not going to be used?

Reply:
You are right in theory. But in practice, there's no code path that does not immediately grab the observable or the promise, so all of the actual heavy work will be triggered no matter what. The only possible savings could be to not create the observable or the promise object itself, saving some minor overhead. But in practice I would argue that it's so minor, that it's not worth worrying about. And remember - a getter in and of itself adds overhead, such that the framework needs to call `Object.defineProperty` and make a function with its closure overhead. Not sure that that's actually measurably cheaper than the Promise creation.